charts/kube-prometheus-stack/values.yaml

```diff
@@ -1597,6 +1597,7 @@ prometheus:
     portName: grpc
     port: 10901
     targetPort: "grpc"
+    omitClusterIP: false
```

Comment:
Why is this needed?
Imho you could just set `clusterIP: ""` to get rid of the error.

Reply:
I'll try that and report back.

Reply:
Same result.
```
UPGRADE FAILED: cannot patch "prometheus-operator-thanos-discovery" with kind Service: Service "prometheus-operator-thanos-discovery" is invalid: spec.clusterIP: Invalid value: "": field is immutable
```